src/Symfony/Component/Translation/Formatter/IntlMessageFormatter.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Formatter;
+
+/**
+ * @author Guilherme Blanco <guilhermeblanco@hotmail.com>
+ * @author Abdellatif Ait boudad <a.aitboudad@gmail.com>
+ */
+class IntlMessageFormatter implements MessageFormatterInterface, ChoiceMessageFormatterInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function format($message, $locale, array $parameters = array())
+    {
+        try {
+            $formatter = new \MessageFormatter($locale, $message);
+        } catch (\Throwable $e) {
+            throw new \InvalidArgumentException('Invalid message format.', $e);
+        }
+        if (null === $formatter) {
+            throw new \InvalidArgumentException(sprintf('Invalid message format. Reason: %s (error #%d)', intl_get_error_message(), intl_get_error_code()));
+        }
+
+        $message = $formatter->format($parameters);
+        if (U_ZERO_ERROR !== $formatter->getErrorCode()) {
+            throw new \InvalidArgumentException(sprintf('Unable to format message. Reason: %s (error #%s)', $formatter->getErrorMessage(), $formatter->getErrorCode()));
+        }
+
+        return $message;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function choiceFormat($message, $number, $locale, array $parameters = array())
+    {
+        return $this->format($message, $locale, $parameters);
```

Comment:
Is this kind of implementation right? Argument `number` is unused.

Reply:
I think we shouldn't implement the `ChoiceMessageFormatterInterface` instead, it will disallow using the `transChoice` method which is right since we should only use the `trans` method for intl-formated message.